packages/jwt-service/src/functions/index.ts

```diff
@@ -418,14 +420,18 @@ export const toJwsJsonGeneralWithIdentifiers = async (
   context: IAgentContext<IIdentifierResolution>
 ): Promise<JwsJsonGeneralWithIdentifiers> => {
   const jws = await toJwsJsonGeneral(args, context)
-  const signatures = await Promise.all(
+  const signatures = (await Promise.all(
     jws.signatures.map(async (signature) => {
       const protectedHeader: JwtHeader = decodeJoseBlob(signature.protected)
       const identifier = args.jwk
         ? await resolveExternalJwkIdentifier({ identifier: args.jwk }, context)
         : await resolveExternalIdentifierFromJwsHeader(protectedHeader, context, args)
-      return { ...signature, identifier }
+      if (identifier !== undefined) {
+        return { ...signature, identifier }
+      }
+      return undefined
     })
-  )
+  )).filter(signature => signature !== undefined ) as JwsJsonSignatureWithIdentifier[]
```

Comment:
This cannot be correct. We now all of a sudden return less objects than we should. Why is this change needed?

Reply:
I am possibly adding undefined elements on line 432. I need to return JwsJsonSignatureWithIdentifier[] , not (JwsJsonSignatureWithIdentifier | undefined)[]

Reply:
Yes, but the whole point is that you started introducing this. If for whatever reason the signature could be empty (which according to the spec it really can't for a jws json general AFAIK), then it still should return the identifiers, not all of a sudden filter everything out. 

But IMO this change reeks of something else going wrong when calling this function. So it should be solved outside of this function. If we really need this then we probably need something like an explicit param that allows the new behaviour (and only when that param is set)

Reply:
resolved